public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -157,11 +158,22 @@ function AddAnomalyDetector({
 
   const notifications = getNotifications();
   const handleValidationAndSubmit = (formikProps) => {
-    if (!isEmpty(formikProps.errors)) {
-      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
-      notifications.toasts.addDanger('One or more input fields is invalid');
+    if (formikProps.values.featureList.length !== 0) {
+      formikProps.setFieldTouched('featureList', true);
+      formikProps.validateForm().then((errors) => {
+        if (!isEmpty(errors)) {
+          focusOnFirstWrongFeature(errors, formikProps.setFieldTouched);
+          notifications.toasts.addDanger(
+            'One or more input fields is invalid.'
+          );
+        } else {
+          handleSubmit(formikProps);
+        }
+      });
     } else {
-      handleSubmit(formikProps);
+      notifications.toasts.addDanger(
+        'One or more features are required to check for anomalies.'
```

Comment:
Can this be simplified to "One or more features are required"? I'm worried the "to check for anomalies" part can add confusion

Reply:
sure, updated it. I added "to check for anomalies" cause that's the wording on existing feature page